src/shared/utils/helpers/is-cake-day.ts

```diff
@@ -1,11 +1,34 @@
-import { parseISO, getYear, getDayOfYear } from "date-fns";
+import { parseISO, getYear, getDayOfYear, isLeapYear } from "date-fns";
+
+const leapDay = getDayOfYear(new Date(2024, 1, 29));
 
 export default function isCakeDay(published: string): boolean {
```

Comment:
I made sure to test that this behaves as described in the PR description. If we weren't pivoting to a rewritten UI, this would be a prime candidate for unit testing.